tools/linker/MobileResolveMainAssemblyStep.cs

```diff
@@ -32,8 +35,16 @@ protected override void Process ()
 			Annotations.Push (assembly);
 
 			foreach (var t in assembly.MainModule.Types) {
-				if (!t.HasCustomAttribute (Namespaces.Foundation, "RegisterAttribute"))
+				if (t.IsPublic && embeddinator) {
+					// Mark all public types when in embeddinator mode.
+					// There may be no types with the Register attribute,
+					// which means that without this the assembly might be completely ignored even if it's not linked.
+					Annotations.Mark (t);
```

Comment:
Yeah, we only need to mark a single thing. I couldn't decide how to select that single thing in any way that made sense, so I just went for something I thought made sense: mark every public type.